inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -702,11 +700,69 @@ Parameter clDNNEngine::GetMetric(const std::string& name, const std::map<std::st
     } else if (name == METRIC_KEY(RANGE_FOR_STREAMS)) {
         std::tuple<unsigned int, unsigned int> range = std::make_tuple(1, 2);
         IE_SET_METRIC_RETURN(RANGE_FOR_STREAMS, range);
+    } else if (name == METRIC_KEY(MAX_BATCH_SIZE)) {
+        const auto& config = _impl->m_configs.GetConfig(device_id);
+        auto n_streams = config.throughput_streams;
+        auto available_device_mem = device_info.max_global_mem_size;
+        int64_t max_batch_size = 0;
+
+        if (options.find("CNN_NETWORK") == options.end()) {
+            throw std::runtime_error("No CNN_NETWORK option given!");
+        }
+        if (options.find("GPU_THROUGHPUT_STREAMS") != options.end()) {
+            n_streams = options.find("GPU_THROUGHPUT_STREAMS")->second.as<uint16_t>();
+        }
+        if (options.find("AVAILABLE_DEVICE_MEM_SIZE") != options.end()) {
+            available_device_mem = options.find("AVAILABLE_DEVICE_MEM_SIZE")->second.as<int64_t>();
+            if (available_device_mem < 0)
+                throw std::runtime_error("No available device mem");
+        }
+
+        auto network = options.find("CNN_NETWORK")->second.as<InferenceEngine::CNNNetwork*>();
```

Comment:
Fixed!

Reply:
@myshevts  It wasnt working because "Parameter::as()"was dyn_cast which cannot change constness. 
To cast with const pointer 1) we need const_cast, or, 2) the original pointer passed from user was const pointer. 
For 1) we'll need to add const_cast for Parameter. 
On the other hand, 2) sounds as expecting too much responsibility for user. 
Or, we can use "non-const" pointer as original.
How do you think?

Reply:
clearly, the 2) as the function promises that the network will not be changed (and e.g. no need to clone tat first).
so user should be passing the const CNNetwork*

Reply:
One problem for 2) is that, the error will be just a blury exception like simple "bad cast", Then the user might be frustrated with that situation. Of course we'll need to describe in the document, but I still have that concern..

Reply:
if user provided the const CNNetwork* (while the code accepts/expects just the "CNNetwork*")  there will be same obsucure "bad cast"- @ilya-lavrenov  we sholud do smarter smarter about it

Reply:
Yes I became to think as adding "const_cast" to the "as" casting is better solution. How do you think?

Reply:
I guess this is right, so that the "as" method should try casting to both const and regular ptr/ref and return whichever succeded, @ilya-lavrenov ?

Reply:
I changed it to pass ngraph::Function* instead of CNNNetwork. 
Also I added some exception comments to show where the casting failed.